src/cargo/util/toml/mod.rs

```diff
@@ -1901,20 +1901,17 @@ fn resolve_source_id_from_dependency<P: ResolveToPath + Clone>(
         orig.registry_index.as_ref(),
     ) {
         (Some(_git), _, Some(_registry), _) | (Some(_git), _, _, Some(_registry)) => bail!(
-            "dependency ({}) specification is ambiguous. \
+            "dependency ({name_in_toml}) specification is ambiguous. \
```

Comment:
aside: isn't there a clippy lint we could turn on for this?

Reply:
Yes, but not sure if we want to turn it on for every place. For here it seems fine.